src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -232,15 +249,25 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
       }),
     });
 
-    // TODO: Analytics
+    analyticsV2.track(analyticsV2.event.swapsSubmitted, {
+      createdAt: Date.now(),
+      type,
+      parameters,
+      selectedGas,
+      slippage,
+      bridge: isBridge,
+      // TODO: Does this work from the JS thread?
```

Comment:
Yes, it does. In general it's likely more efficient to read from the shared values than from the store. You can always read a shared value from either thread, changes in the value just don't trigger re-renders. On the JS thread they behave just like refs.